File: lib/AdditionalCellState/AdditionalCellStateShowEditing.m
//
//  AdditionalCellStateShowEditing.m
//  TableViewAgent
//
//  Created by P.I.akura on 2013/06/18.
//  Copyright (c) 2013年 P.I.akura. All rights reserved.
//

#import "AdditionalCellStateShowEditing.h"

@implementation AdditionalCellStateShowEditing

- (BOOL)isShowAddCell:(BOOL) editing {
    return editing;
}

- (ChangeInState)changeInState:(BOOL) editing {
    return (editing == NO) ? ChangeInStateHide : ChangeInStateShow;
}

@end

File: lib/AdditionalCellState/AdditionalCellStateShowEditing.swift
//
//  AdditionalCellStateShowEditing.swift
//  TableViewAgent
//
//  Created by akuraru on 2014/07/29.
//  Copyright (c) 2014年 P.I.akura. All rights reserved.
//

import Foundation

class AdditionalCellStateShowEditing : AdditionalCellState {
    override func isShowAddCell(editing :Bool) -> Bool {
        return editing
    }
    
    override func changeInState(editing :Bool) -> Int {
        return (editing == false) ? 1 : 2;
    }
}